### ISSUE TITLE
Pondering in GUI

### ISSUE THREAD

**Issue by rwst** [cutechess/cutechess#64]
Apparently the Init String field in the GUI cannot be used to set ponder.

Previous description:
How is ponder supposed to work? In a game with stockfish and ponder set, top shows only the current engine taking CPU/memory. If there is a misunderstanding, the documentation should clarify better. If stockfish does not support ponder (anymore) it should be clarified there.

**Comment by ilaripih** [comment]
I don't know if the late versions of Stockfish have made some changes to pondering, but I just tried it with cutechess-cli 0.8.0 and Stockfish 261014 and it worked just fine with this command:
`./cutechess-cli -engine conf=Stockfish name=stock1 -engine conf=Stockfish name=stock2 -each tc=60 ponder -debug`

Here's some of the output from the beginning:
```
3 >stock1(0): uci
7 >stock2(1): uci
8 <stock1(0): Stockfish 261014 64 by Tord Romstad, Marco Costalba and Joona Kiiski
8 <stock2(1): Stockfish 261014 64 by Tord Romstad, Marco Costalba and Joona Kiiski
98 <stock1(0): id name Stockfish 261014 64
98 <stock1(0): id author Tord Romstad, Marco Costalba and Joona Kiiski
98 <stock1(0): option name Write Debug Log type check default false
98 <stock1(0): option name Write Search Log type check default false
98 <stock1(0): option name Search Log Filename type string default SearchLog.txt
98 <stock1(0): option name Contempt Factor type spin default 0 min -50 max 50
98 <stock1(0): option name Min Split Depth type spin default 0 min 0 max 12
98 <stock1(0): option name Threads type spin default 1 min 1 max 128
98 <stock1(0): option name Hash type spin default 16 min 1 max 65536
98 <stock1(0): option name Clear Hash type button
98 <stock1(0): option name Ponder type check default true
98 <stock1(0): option name MultiPV type spin default 1 min 1 max 500
98 <stock1(0): option name Skill Level type spin default 20 min 0 max 20
98 <stock1(0): option name Emergency Move Horizon type spin default 40 min 0 max 50
98 <stock1(0): option name Emergency Base Time type spin default 60 min 0 max 30000
98 <stock1(0): option name Emergency Move Time type spin default 30 min 0 max 5000
98 <stock1(0): option name Minimum Thinking Time type spin default 20 min 0 max 5000
98 <stock1(0): option name Slow Mover type spin default 80 min 10 max 1000
98 <stock1(0): option name UCI_Chess960 type check default false
98 <stock1(0): uciok
98 >stock1(0): setoption name Clear Hash
98 >stock1(0): setoption name Contempt Factor value 0
98 >stock1(0): setoption name Emergency Base Time value 60
98 >stock1(0): setoption name Emergency Move Horizon value 40
98 >stock1(0): setoption name Emergency Move Time value 30
98 >stock1(0): setoption name Hash value 16
98 >stock1(0): setoption name Min Split Depth value 0
98 >stock1(0): setoption name Minimum Thinking Time value 20
98 >stock1(0): setoption name MultiPV value 1
98 >stock1(0): setoption name Search Log Filename value SearchLog.txt
98 >stock1(0): setoption name Skill Level value 20
98 >stock1(0): setoption name Slow Mover value 80
98 >stock1(0): setoption name Threads value 1
98 >stock1(0): setoption name Write Debug Log value false
98 >stock1(0): setoption name Write Search Log value false
98 >stock1(0): isready
98 <stock2(1): id name Stockfish 261014 64
98 <stock2(1): id author Tord Romstad, Marco Costalba and Joona Kiiski
98 <stock2(1): option name Write Debug Log type check default false
98 <stock2(1): option name Write Search Log type check default false
98 <stock2(1): option name Search Log Filename type string default SearchLog.txt
98 <stock2(1): option name Contempt Factor type spin default 0 min -50 max 50
98 <stock2(1): option name Min Split Depth type spin default 0 min 0 max 12
98 <stock2(1): option name Threads type spin default 1 min 1 max 128
98 <stock2(1): option name Hash type spin default 16 min 1 max 65536
98 <stock2(1): option name Clear Hash type button
98 <stock2(1): option name Ponder type check default true
98 <stock2(1): option name MultiPV type spin default 1 min 1 max 500
98 <stock2(1): option name Skill Level type spin default 20 min 0 max 20
99 <stock2(1): option name Emergency Move Horizon type spin default 40 min 0 max 50
99 <stock2(1): option name Emergency Base Time type spin default 60 min 0 max 30000
99 <stock2(1): option name Emergency Move Time type spin default 30 min 0 max 5000
99 <stock2(1): option name Minimum Thinking Time type spin default 20 min 0 max 5000
99 <stock2(1): option name Slow Mover type spin default 80 min 10 max 1000
99 <stock2(1): option name UCI_Chess960 type check default false
99 <stock2(1): uciok
99 >stock2(1): setoption name Clear Hash
99 >stock2(1): setoption name Contempt Factor value 0
99 >stock2(1): setoption name Emergency Base Time value 60
99 >stock2(1): setoption name Emergency Move Horizon value 40
99 >stock2(1): setoption name Emergency Move Time value 30
99 >stock2(1): setoption name Hash value 16
99 >stock2(1): setoption name Min Split Depth value 0
99 >stock2(1): setoption name Minimum Thinking Time value 20
99 >stock2(1): setoption name MultiPV value 1
99 >stock2(1): setoption name Search Log Filename value SearchLog.txt
99 >stock2(1): setoption name Skill Level value 20
99 >stock2(1): setoption name Slow Mover value 80
99 >stock2(1): setoption name Threads value 1
99 >stock2(1): setoption name Write Debug Log value false
99 >stock2(1): setoption name Write Search Log value false
99 >stock2(1): isready
103 <stock1(0): readyok
103 <stock2(1): readyok
Started game 1 of 1 (stock1 vs stock2)
103 >stock1(0): ucinewgame
103 >stock1(0): setoption name Ponder value true
103 >stock1(0): position startpos
103 >stock2(1): ucinewgame
103 >stock2(1): setoption name Ponder value true
103 >stock2(1): position startpos
103 >stock1(0): isready
107 <stock1(0): readyok
107 >stock1(0): go wtime 60000 btime 60000
107 <stock1(0): info depth 1 seldepth 1 score cp 85 nodes 27 nps 27000 time 1 multipv 1 pv e2e4
107 <stock1(0): info depth 2 seldepth 2 score cp 6 nodes 155 nps 155000 time 1 multipv 1 pv d2d4 d7d5
108 <stock1(0): info depth 3 seldepth 3 score cp 64 nodes 386 nps 386000 time 1 multipv 1 pv d2d4 d7d5 g1f3
108 <stock1(0): info depth 4 seldepth 4 score cp 25 nodes 1082 nps 541000 time 2 multipv 1 pv e2e4 e7e5 d2d4 b8c6
109 <stock1(0): info depth 5 seldepth 5 score cp 53 nodes 2997 nps 999000 time 3 multipv 1 pv b1c3 g8f6 d2d4 d7d5 g1f3
110 <stock1(0): info depth 6 seldepth 6 score cp 26 nodes 4557 nps 1519000 time 3 multipv 1 pv e2e4 e7e5 d2d4 g8f6 b1c3 b8c6
111 <stock1(0): info depth 7 seldepth 7 score cp 58 nodes 6692 nps 1673000 time 4 multipv 1 pv e2e4 d7d5 e4e5 b8c6 d2d4 e7e6 f1b5
114 <stock1(0): info depth 8 seldepth 8 score cp 10 nodes 14330 nps 1791250 time 8 multipv 1 pv e2e4 d7d5 e4d5 g8f6 d2d4 f6d5 c2c4 d5b4
119 <stock1(0): info depth 9 seldepth 11 score cp 33 nodes 23942 nps 1841692 time 13 multipv 1 pv e2e4 d7d5 e4e5 c7c5 d2d4 c5d4 d1d4 b8c6 f1b5 d8a5 b1c3 a5b5 c3b5 c6d4 b5d4
134 <stock1(0): info depth 10 seldepth 14 score cp 40 nodes 51624 nps 1843714 time 28 multipv 1 pv e2e4 b8c6 b1c3 g8f6 g1f3 d7d5 e4e5 f6e4 d2d4 e7e6
155 <stock1(0): info depth 11 seldepth 15 score cp 28 nodes 89188 nps 1858083 time 48 multipv 1 pv e2e4 b8c6 b1c3 g8f6 d2d4 d7d5 e4e5 f6e4 g1f3 e7e6 f1b5 f8b4 e1g1 e4c3 b2c3 b4c3
164 <stock1(0): info depth 12 seldepth 15 score cp 28 nodes 106341 nps 1833465 time 58 multipv 1 pv e2e4 b8c6 b1c3 g8f6 d2d4 d7d5 e4e5 f6e4 g1f3 e7e6 f1b5 f8b4 e1g1 e4c3 b2c3 b4c3
269 <stock1(0): info depth 13 seldepth 18 score cp 38 nodes 287616 nps 1764515 time 163 multipv 1 pv e2e4 d7d5 e4d5 g8f6 d2d4 e7e6 d5e6 c8e6 g1f3 b8c6 c2c3 f8e7 c1e3
505 <stock1(0): info depth 14 seldepth 20 score cp 26 nodes 687159 nps 1722203 time 399 multipv 1 pv g1f3 d7d5 d2d4 e7e6 c1f4 g8f6 b1d2 b8c6 e2e3 f8e7 f1d3 c8d7 e1g1 f6h5
597 <stock1(0): info depth 15 seldepth 20 score cp 26 nodes 837927 nps 1706572 time 491 multipv 1 pv g1f3 d7d5 d2d4 e7e6 c1f4 g8f6 b1d2 b8c6 e2e3 f8d6
773 <stock1(0): info depth 16 seldepth 20 score cp 25 nodes 1130137 nps 1694358 time 667 multipv 1 pv e2e4 e7e5 b1c3 g8f6 g1f3 b8c6 f1b5 f8b4 e1g1 e8g8 d2d3 d7d6 b5c6 b7c6 c1e3 b4c3 b2c3
1553 <stock1(0): info depth 17 seldepth 23 score cp 25 nodes 2351793 nps 1625288 time 1447 multipv 1 pv e2e4 e7e5 g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1724 <stock1(0): info depth 18 seldepth 23 score cp 25 nodes 2629450 nps 1625123 time 1618 multipv 1 pv e2e4 e7e5 g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1724 <stock1(0): info nodes 2629450 time 1618
1724 <stock1(0): bestmove e2e4 ponder e7e5
1724 >stock2(1): position startpos moves e2e4
1724 >stock2(1): isready
1724 >stock1(0): position startpos moves e2e4 e7e5
1724 >stock1(0): isready
1725 <stock1(0): readyok
1725 >stock1(0): go ponder wtime 58383 btime 60000
1725 <stock2(1): readyok
1725 >stock2(1): go wtime 58383 btime 60000
1725 <stock1(0): info depth 1 seldepth 1 score cp 91 nodes 81 nps 40500 time 2 multipv 1 pv d2d4 e5d4
1725 <stock1(0): info depth 2 seldepth 2 score cp 31 nodes 196 nps 98000 time 2 multipv 1 pv d2d4 g8f6 d4e5
1725 <stock2(1): info depth 1 seldepth 1 score cp 0 nodes 45 nps 22500 time 2 multipv 1 pv d7d5
1725 <stock1(0): info depth 3 seldepth 3 score cp 25 nodes 483 nps 241500 time 2 multipv 1 pv d2d4 e5d4 g1f3 c7c5 f1c4 b8c6 e1g1 d8e7
1725 <stock2(1): info depth 2 seldepth 2 score cp -65 nodes 199 nps 99500 time 2 multipv 1 pv d7d5 b1c3 d5e4 c3e4
1725 <stock1(0): info depth 4 seldepth 4 score cp 25 nodes 764 nps 382000 time 2 multipv 1 pv d2d4 e5d4 g1f3 c7c5 f1c4 b8c6 e1g1 d8e7
1725 <stock2(1): info depth 3 seldepth 3 score cp -2 nodes 466 nps 233000 time 2 multipv 1 pv d7d5 b1c3 g8f6
1725 <stock1(0): info depth 5 seldepth 5 score cp 25 nodes 1236 nps 618000 time 2 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1725 <stock1(0): info depth 6 seldepth 5 score cp 25 nodes 1341 nps 670500 time 2 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1726 <stock1(0): info depth 7 seldepth 5 score cp 25 nodes 1504 nps 752000 time 2 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1726 <stock1(0): info depth 8 seldepth 5 score cp 25 nodes 1756 nps 878000 time 2 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1726 <stock1(0): info depth 9 seldepth 5 score cp 25 nodes 2039 nps 679666 time 3 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1726 <stock2(1): info depth 4 seldepth 4 score cp -20 nodes 1214 nps 404666 time 3 multipv 1 pv d7d5 b1c3 g8f6 e4e5
1726 <stock1(0): info depth 10 seldepth 5 score cp 25 nodes 2590 nps 863333 time 3 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1726 <stock1(0): info depth 11 seldepth 5 score cp 25 nodes 3177 nps 1059000 time 3 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1727 <stock1(0): info depth 12 seldepth 5 score cp 25 nodes 3980 nps 1326666 time 3 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1727 <stock2(1): info depth 5 seldepth 5 score cp -6 nodes 2920 nps 730000 time 4 multipv 1 pv b8c6 b1c3 e7e5 g1f3 g8f6
1728 <stock2(1): info depth 6 seldepth 6 score cp -53 nodes 4812 nps 962400 time 5 multipv 1 pv b8c6 b1c3 g8f6 d2d4 d7d5 e4e5
1728 <stock1(0): info depth 13 seldepth 5 score cp 25 nodes 6129 nps 1225800 time 5 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1731 <stock2(1): info depth 7 seldepth 7 score cp -6 nodes 10808 nps 1351000 time 8 multipv 1 pv d7d5 e4d5 g8f6 b1c3 f6d5 d2d4 b8c6
1731 <stock1(0): info depth 14 seldepth 5 score cp 25 nodes 12027 nps 1503375 time 8 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1737 <stock2(1): info depth 8 seldepth 9 score cp -40 nodes 22204 nps 1586000 time 14 multipv 1 pv d7d5 e4e5 c7c5 d2d4 b8c6 c2c3 c5d4 c3d4
1749 <stock1(0): info depth 15 seldepth 5 score cp 25 nodes 40632 nps 1562769 time 26 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1750 <stock2(1): info depth 9 seldepth 13 score cp -24 nodes 45256 nps 1676148 time 27 multipv 1 pv e7e5 b1c3 b8c6 g1f3 g8f6 d2d4 e5d4 f3d4 d7d5
1764 <stock2(1): info depth 10 seldepth 13 score cp -36 nodes 70075 nps 1751875 time 40 multipv 1 pv e7e5 b1c3 b8c6 g1f3 g8f6 d2d4 e5d4 f3d4 f8b4 c1g5
1778 <stock1(0): info depth 16 seldepth 5 score cp 25 nodes 88011 nps 1600200 time 55 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 c1e3 c8g4 a2a3 c5e3 f2e3
1786 <stock2(1): info depth 11 seldepth 14 score cp -31 nodes 108470 nps 1749516 time 62 multipv 1 pv e7e5 b1c3 b8c6 f1c4 g8f6 d2d3 f8c5 g1f3 e8g8 e1g1 d7d6 c1d2
1875 <stock2(1): info depth 12 seldepth 17 score cp -22 nodes 244482 nps 1608434 time 152 multipv 1 pv e7e5 b1c3 g8f6 g1f3 b8c6 f1b5 f8b4 d2d3 e8g8 e1g1 d7d6 c1e3 a7a6 b5c6 b7c6
1908 <stock2(1): info depth 13 seldepth 17 score cp -22 nodes 295904 nps 1599481 time 185 multipv 1 pv e7e5 b1c3 g8f6 g1f3 b8c6 f1b5 f8b4 d2d3 e8g8 e1g1 d7d6 c1e3 a7a6 b5c6 b7c6
2174 <stock2(1): info depth 14 seldepth 18 score cp -24 nodes 708319 nps 1574042 time 450 multipv 1 pv e7e5 g1f3 b8c6 f1b5 c6d4 b1c3 d4f3 d1f3 g8f6 b5e2 c7c6 d2d3 b7b5 f3g3 d7d6
2198 <stock1(0): info depth 17 seldepth 24 score cp 27 nodes 706008 nps 1486332 time 475 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 h2h3 c8d7 a2a3 a7a6 c1e3 c5e3 f2e3 c6c5 c3d5 f6d5 e4d5
2303 <stock2(1): info depth 15 seldepth 18 score cp -23 nodes 908659 nps 1566653 time 580 multipv 1 pv e7e5 g1f3 b8c6 f1b5 c6d4 b1c3 d4f3 d1f3 g8f6 f3g3 c7c6 b5e2 d8e7 e1g1 d7d5 d2d3 d5d4
2488 <stock2(1): info depth 16 seldepth 21 score cp -27 nodes 1193535 nps 1562218 time 764 multipv 1 pv e7e5 g1f3 b8c6 f1b5 c6d4 f3d4 e5d4 b5c4 d7d5 e4d5 d8e7 d1e2 e7e2 e1e2 g8f6 d2d3 f8d6 b1d2 e8g8
2811 <stock1(0): info depth 18 seldepth 24 score cp 27 nodes 1622565 nps 1491328 time 1088 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 h2h3 c8d7 a2a3 a7a6 c1e3 c5e3 f2e3 c6c5 c3d5 f6d5 e4d5
3109 <stock2(1): info depth 17 seldepth 25 score cp -32 nodes 2145871 nps 1548247 time 1386 multipv 1 pv e7e5 g1f3 b8c6 f1b5 c6d4 f3d4 e5d4 e1g1 c7c6 b5d3 d7d5 f1e1 g8e7 e4d5 d8d5
3109 <stock2(1): info nodes 2145871 time 1386
3109 <stock2(1): bestmove e7e5 ponder g1f3
3110 >stock1(0): ponderhit
3110 >stock2(1): position startpos moves e2e4 e7e5 g1f3
3110 >stock2(1): isready
3110 <stock1(0): info depth 19 seldepth 24 score cp 27 nodes 2054282 nps 1482165 time 1386 multipv 1 pv g1f3 b8c6 f1b5 g8f6 e1g1 f8c5 b5c6 d7c6 d2d3 d8e7 b1c3 e8g8 h2h3 c8d7 a2a3 a7a6 c1e3 c5e3 f2e3 c6c5 c3d5 f6d5 e4d5
3110 <stock1(0): info nodes 2054282 time 1386
3110 <stock1(0): bestmove g1f3 ponder b8c6
3110 >stock1(0): position startpos moves e2e4 e7e5 g1f3 b8c6
3110 >stock1(0): isready
3110 <stock2(1): readyok
3110 >stock2(1): go ponder wtime 58383 btime 58615
3110 >stock2(1): ponderhit
3110 <stock1(0): readyok
3110 >stock1(0): go ponder wtime 58383 btime 58615
3111 <stock2(1): info depth 1 seldepth 1 score cp -15 nodes 86 nps 86000 time 1 multipv 1 pv d7d5 e4d5 e5e4
3111 <stock2(1): info depth 2 seldepth 2 score cp -15 nodes 154 nps 154000 time 1 multipv 1 pv d7d5 e4d5 e5e4
3111 <stock1(0): info depth 1 seldepth 1 score cp 81 nodes 90 nps 90000 time 1 multipv 1 pv d2d4 e5d4 f3d4 g8f6
```

Can you try the same command, capture the debug output and paste it here?

**Comment by rwst** [comment]
I confirm that it works on CLI but <pparently the Init String field in the GUI cannot be used to set ponder.

I used "ponder" and "engine ponder" there.

I changed the issue description accordingly.

**Comment by ilaripih** [comment]
Ah yes, that won't work. You can send anything you want to the engine as an "init string" but that doesn't make CuteChess send the "go ponder" command to UCI engine. So at the moment, pondering is not supported in the GUI side.

**Comment by ilaripih** [comment]
Pondering should be an engine option that can only be set temporarily for a game or a tournament, ie. it should not be configurable via `SettingsDialog`.

**Comment by ilaripih** [comment]
The pondering option will be implemented as a part of #82. I'm closing this as a duplicate.
